contracts/identity/Identity.sol

```diff
@@ -79,13 +76,17 @@ contract Identity is ProxyStorage {
             lastNonce++;
         }
 
-        bool status = applyOperation(to, value, data, operationType);
-        emit TransactionExecution(hash, status);
+        uint startGas = gasleft();
+        require(startGas >= gasLimit, "Not enough gas left for operation");
```

Comment:
I don't think that's correct. The user could set a super high gas limit of 4million but the tx only costs 100k and the delegate gives 100k gas, so `startGas < gasLimit` but the tx is still valid.

Reply:
mmh but without this check a delegate could easily make a transaction fail just by giving not enough gas.

Reply:
I think the delegate should always use `gas = userGaslimit + expectedOverheadGas` or reject the meta tx.

Reply:
Otherwise their is now way for me as a user to tell the delegate that this transaction will need 4M gas.